keylime/common/algorithms.py

```diff
@@ -13,17 +17,44 @@ def is_accepted(algorithm, accepted):
     return algorithm in accepted
 
 
-class Hash:
+class Hash(str, enum.Enum):
     SHA1 = 'sha1'
     SHA256 = 'sha256'
     SHA384 = 'sha384'
     SHA512 = 'sha512'
     SM3_256 = 'sm3_256'
-    supported_algorithms = (SHA1, SHA256, SHA384, SHA512, SM3_256)
 
     @staticmethod
     def is_recognized(algorithm):
-        return algorithm in Hash.supported_algorithms
+        try:
+            Hash(algorithm)
+            return True
+        except ValueError:
+            return False
+
+    def hash(self, data: bytes) -> Optional[bytes]:
+        if self == Hash.SHA1:
+            return hashlib.sha1(data).digest()
+        if self == Hash.SHA256:
+            return hashlib.sha256(data).digest()
+        if self == Hash.SHA384:
+            return hashlib.sha384(data).digest()
+        if self == Hash.SHA512:
+            return hashlib.sha512(data).digest()
+        if self == Hash.SM3_256:
+            # SM3 might not be guaranteed to be there
+            try:
+                return hashlib.new("sm3", data).digest()
+            except ValueError:
+                return None
+
+        return None
+
+    def get_size(self):
+        return _HASH_SIZE.get(self)
```

Comment:
SM3 is not listed below in _HASH_SIZE...

Reply:
fixed